cmd/.clangd

```diff
@@ -0,0 +1,2 @@
+CompileFlags:
+  Add: -I/usr/include/glib-2.0 -Wall -Wextra -Wmissing-prototypes -Wstrict-prototypes -Wno-missing-field-initializers -Wno-unused-parameter
```

Comment:
This looks super interessting, how do you use this?

Reply:
I'm using spacemacs with language server protocol setup. That in turn uses clangd to provide completions, indexing and a little bit of refactoring as well as built in annotations with warnings.

Reply:
Hmm I am using normal emacs with lsp-mode etc and clangd and this seems to work reasonably well - one thing I did notice was it doesn't define HAVE_CONFIG_H and so we don't ever include config.h - which in turns means things like `_GNU_SOURCE` never get defined and so I get errors like the following:

```
 snap-confine.c    66  24 error    undeclared_var_use Use of undeclared identifier 'O_PATH' (lsp)
 snap-confine.c   146   6 error    undeclared_var_use Use of undeclared identifier 'O_PATH' (lsp)
 snap-confine.c   209  21 error    undeclared_var_use Use of undeclared identifier 'O_PATH' (lsp)
 snap-confine.c   267  21 error    undeclared_var_use Use of undeclared identifier 'O_PATH' (lsp)
 snap-confine.c   277  22 error    undeclared_var_use Use of undeclared identifier 'O_PATH' (lsp)
 snap-confine.c   326  21 error    undeclared_var_use Use of undeclared identifier 'PACKAGE' (lsp)
 snap-confine.c   326  30 error    undeclared_var_use Use of undeclared identifier 'PACKAGE_VERSION' (lsp)
 snap-confine.c   343   6 warning  -Wimplicit-function-declaration Implicit declaration of function 'getresuid' is invalid in C99 (lsp)
 snap-confine.c   346   6 warning  -Wimplicit-function-declaration Implicit declaration of function 'getresgid' is invalid in C99 (lsp)
 snap-confine.c   588   6 warning  -Wimplicit-function-declaration Implicit declaration of function 'unshare' is invalid in C99 (lsp)
 snap-confine.c   588  14 error    undeclared_var_use Use of undeclared identifier 'CLONE_NEWNS' (lsp)
 snap-confine.c   672   7 warning  -Wimplicit-function-declaration Implicit declaration of function 'unshare' is invalid in C99 (lsp)
 snap-confine.c   672  15 error    undeclared_var_use Use of undeclared identifier 'CLONE_NEWNS' (lsp)
 snap-confine.c   696   8 warning  -Wimplicit-function-declaration Implicit declaration of function 'unshare' is invalid in C99 (lsp)
 snap-confine.c   696  16 error    undeclared_var_use Use of undeclared identifier 'CLONE_NEWNS' (lsp)
```

Adding `-D_GNU_SOURCE` to `.clangd` seems to resolve most of these but I wonder if there may be a better way (or perhaps I just have something misconfigured on my end)?

Reply:
Ha, good catch, I was wondering why it highlighted those.

Reply:
So I found that I have some better success if I instead use a `compile_flags.txt` in place of `.clangd` with the following contents (for some reason if I just added the same entries to .clangd it doesn't seem to work):

```
-I/usr/include/glib-2.0
-Wall
-Wextra
-Wmissing-prototypes
-Wstrict-prototypes
-Wno-missing-field-initializers
-Wno-unused-parameter
-DHAVE_CONFIG_H=1
-I./
```

In this case we get to make use of config.h (but this then relies on a user having run configure previously to generate it)...

Reply:
btw. what's your clangd version?

Reply:
`Ubuntu clangd version 12.0.0-1ubuntu1` (clang-12 on hirsute)